src/components/accessories/patientTherapy/PatientTherapy.tsx

```diff
@@ -4,123 +4,139 @@ import TherapyForm from "./therapyForm/TherapyForm";
-  }, [getMedicals]);
+    dispatch(createTherapyReset());
+    dispatch(updateTherapyReset());
+  }, [dispatch, updateTherapyReset, createTherapyReset]);
 
   useEffect(() => {
     if (activityTransitionState === "TO_RESET") {
-      createTherapyReset();
-      setShouldResetForm(true);
       setShouldUpdateTable(true);
+      setCreationMode(true);
+      dispatch(createTherapyReset());
+      dispatch(updateTherapyReset());
+      setShouldResetForm(true);
     }
-  }, [activityTransitionState, createTherapyReset]);
-
-  useEffect(() => {
-    getTherapiesByPatientId(patientData?.code);
-  }, [patientData, getTherapiesByPatientId]);
+  }, [
+    dispatch,
+    activityTransitionState,
+    createTherapyReset,
+    updateTherapyReset,
+  ]);
 
-  const onSubmit = (therapy: TherapyRowDTO) => {
+  const onSubmit = (valuesToSave: TherapyRowDTO) => {
     setShouldResetForm(false);
-    therapy.patID = patientData;
-    createTherapy(therapy);
+    valuesToSave.therapyID = therapyToEdit.therapyID;
+    valuesToSave.patID = patientData;
+    if (!creationMode && therapyToEdit.therapyID) {
+      dispatch(updateTherapy(valuesToSave));
+    } else dispatch(createTherapy(valuesToSave));
   };
 
   const resetFormCallback = () => {
     setShouldResetForm(false);
-    setShouldUpdateTable(false);
+    setCreationMode(true);
+    dispatch(createTherapyReset());
+    dispatch(updateTherapyReset());
     setActivityTransitionState("IDLE");
+    setShouldUpdateTable(false);
+    scrollToElement(null);
+  };
+
+  const onEdit = (row: TherapyRowDTO) => {
+    setTherapyToEdit(row);
+    setCreationMode(false);
     scrollToElement(null);
   };
 
   return (
     <div className="patientTherapy">
       <TherapyForm
-        fields={initialFields}
+        fields={
+          creationMode
+            ? initialFields
+            : updateFields(initialFields, therapyToEdit)
+        }
         onSubmit={onSubmit}
-        submitButtonLabel={t("common.savetherapy")}
+        submitButtonLabel={
+          creationMode ? t("therapy.savetherapy") : t("therapy.updatetherapy")
+        }
         resetButtonLabel={t("common.discard")}
         shouldResetForm={shouldResetForm}
         resetFormCallback={resetFormCallback}
-        isLoading={isLoading}
+        isLoading={createStatus === "LOADING" || updateStatus === "LOADING"}
       />
-      <div ref={infoBoxRef} className="info-box-container">
-        {hasFailed && (
-          <InfoBox
-            type="error"
-            message="Something went wrong, please retry later."
-          />
-        )}
-      </div>
+      {(createStatus === "FAIL" || updateStatus === "FAIL") && (
+        <div ref={infoBoxRef} className="info-box-container">
+          <InfoBox type="error" message={t("common.somethingwrong")} />
+        </div>
+      )}
+
       <ConfirmationDialog
-        isOpen={hasSucceeded}
-        title="Therapy Created"
+        isOpen={createStatus === "SUCCESS" || updateStatus === "SUCCESS"}
```

Comment:
I would handler this behaviour with a useState: use an useEffect in order to properly update the state once one of this variables will change, do you agree?